test/tactics.lean

```diff
@@ -129,57 +130,72 @@ end
 
 end h_generalize
 
--- section tfae
-
--- example (p q r s : Prop)
---   (h₀ : p ↔ q)
---   (h₁ : q ↔ r)
---   (h₂ : r ↔ s) :
---   p ↔ s :=
--- begin
---   scc,
--- end
-
--- example (p' p q r r' s s' : Prop)
---   (h₀ : p' → p)
---   (h₀ : p → q)
---   (h₁ : q → r)
---   (h₁ : r' → r)
---   (h₂ : r ↔ s)
---   (h₂ : s → p)
---   (h₂ : s → s') :
---   p ↔ s :=
--- begin
---   scc,
--- end
-
--- example (p' p q r r' s s' : Prop)
---   (h₀ : p' → p)
---   (h₀ : p → q)
---   (h₁ : q → r)
---   (h₁ : r' → r)
---   (h₂ : r ↔ s)
---   (h₂ : s → p)
---   (h₂ : s → s') :
---   p ↔ s :=
--- begin
---   scc',
---   assumption
--- end
-
--- example : tfae [true, ∀ n : ℕ, 0 ≤ n * n, true, true] := begin
---   tfae_have : 3 → 1, { intro h, constructor },
---   tfae_have : 2 → 3, { intro h, constructor },
---   tfae_have : 2 ← 1, { intros h n, apply nat.zero_le },
---   tfae_have : 4 ↔ 2, { tauto },
---   tfae_finish,
--- end
-
--- example : tfae [] := begin
---   tfae_finish,
--- end
-
--- end tfae
+section tfae
+
+example (p q r s : Prop)
+  (h₀ : p ↔ q)
+  (h₁ : q ↔ r)
+  (h₂ : r ↔ s) :
+  p ↔ s :=
+begin
+  scc,
+end
+
+example (p' p q r r' s s' : Prop)
+  (h₀ : p' → p)
+  (h₀ : p → q)
+  (h₁ : q → r)
+  (h₁ : r' → r)
+  (h₂ : r ↔ s)
+  (h₂ : s → p)
+  (h₂ : s → s') :
+  p ↔ s :=
+begin
+  scc,
+end
+
+example (p' p q r r' s s' : Prop)
+  (h₀ : p' → p)
+  (h₀ : p → q)
+  (h₁ : q → r)
+  (h₁ : r' → r)
+  (h₂ : r ↔ s)
+  (h₂ : s → p)
+  (h₂ : s → s') :
+  p ↔ s :=
+begin
+  scc',
+  assumption
+end
+
+example : tfae [true, ∀ n : ℕ, 0 ≤ n * n, true, true] := begin
+  tfae_have : 3 → 1, { intro h, constructor },
+  tfae_have : 2 → 3, { intro h, constructor },
+  tfae_have : 2 ← 1, { intros h n, apply nat.zero_le },
+  tfae_have : 4 ↔ 2, { tauto },
+  tfae_finish,
+end
+
+example : tfae [] := begin
+  tfae_finish,
+end
+
+variables P Q R : Prop
+
+example : tfae [P, Q, R] :=
+begin
+  have : P → Q := sorry, have : Q → R := sorry, have : R → P := sorry,
+  --have : R → Q := sorry, -- uncommenting this makes the proof fail
+  tfae_finish
+end
+
+example : tfae [P, Q, R] :=
+begin
+  have : P ↔ Q := sorry, have : Q ↔ R := sorry,
+  tfae_finish -- the success or failure of this tactic is nondeterministic!
```

Comment:
Should these comments remain here? And the sorry's?

Reply:
I'm hoping these two examples with `sorry`s were tests you added before fixing `tfae`, right?!

Reply:
@rwbarton No, they are the examples provided in #1559 that I added as a regression test. 

@jcommelin I'll remove the comments / uncomment the triggering lines.